src/main/java/org/spongepowered/api/world/storage/WorldStorage.java

```diff
@@ -24,12 +24,58 @@
  */
 package org.spongepowered.api.world.storage;
 
+import com.flowpowered.math.vector.Vector3i;
+import com.google.common.base.Optional;
+import org.spongepowered.api.service.persistence.data.DataContainer;
+import org.spongepowered.api.world.Chunk;
+
+import java.util.concurrent.Future;
+
 public interface WorldStorage {
 
     /**
-     * Gets a {@link ChunkIterator}.
+     * Gets a {@link ChunkDataStream}.
+     *
+     * <p>Usage of a {@link ChunkDataStream} should be limited to asynchronous
+     * tasks to avoid hanging the main thread.</p>
+     *
      * @return An iterator of generated chunks
      */
-    ChunkIterator getGeneratedChunks();
+    ChunkDataStream getGeneratedChunks();
+
+    /**
+     * Checks if the given chunk coordinates represented by {@link Vector3i}
+     * exist in the world.
+     *
+     * <p>Note that this is an asynchronous check as the storage of chunks
+     * can not be guaranteed to remain in synch with the server, let alone
+     * on the server thread.</p>
+     *
+     * <p>It is imperative to avoid waiting for the {@link Future} to complete
+     * its task.</p>
```

Comment:
javadocs could be clarified but this is to stop the mass of plugin developers deciding to suspend the main thread waiting for a future.